File: Day54/DSA/Most_Frequent_Even_Element.java
Question:-  https://leetcode.com/problems/most-frequent-even-element/description/

Code:-

class Solution {
    public int mostFrequentEven(int[] nums) {
        ArrayList<Integer> list = new ArrayList<>();

        HashMap<Integer, Integer> mp = new HashMap<>();

        for (int i : nums) 
        {
            if (i % 2 == 0) 
            {
                list.add(i);
            }
        }

        for(int i : list)
        {
            int count = mp.getOrDefault(i,0);
            mp.put(i , count + 1);
        }

        int mostfrequent = 0;
        int element = -1;

        for(Map.Entry<Integer,Integer> ele : mp.entrySet())
        {
            if(ele.getValue() > mostfrequent)
            {
                mostfrequent = ele.getValue();
                element = ele.getKey();
            }
            else if(ele.getValue() == mostfrequent && ele.getKey() < element)
            {
                element = ele.getKey();
            } 
        }

        return element;


    }
}

File: Day54/DSA/Most_Frequent_Even_Element.cpp
Question:-  https://leetcode.com/problems/most-frequent-even-element/description/


Code:-

class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        vector<int> even;

        for (auto i : nums) {
            if (i % 2 == 0) {
                even.push_back(i);
            }
        }

        unordered_map<int, int> mp;
        for (auto i : even) {
            mp[i]++;
        }

        int mostfrequent = 0;
        int ele = -1;
        for (auto i : mp) {
            if (i.second > mostfrequent) {
                mostfrequent = i.second;
                ele = i.first;
            } else if (i.second == mostfrequent && i.first < ele) {
                ele = i.first;
            }
        }

        return ele;
    }
};